src/NServiceBus.Persistence.DynamoDB/DynamoDbInstaller.cs

```diff
@@ -3,19 +3,33 @@
     using System.Threading;
     using System.Threading.Tasks;
     using Installation;
+    using Settings;
 
     /// <summary>
     /// This is a wrapper since NServiceBus always registers implementations of <see cref="INeedToInstallSomething"/> in DI which means that all ctor arguments need to be resolvable via DI too. Instead, the actual, preconfigured, installer will be injected to simplify DI configuration.
     /// </summary>
     class DynamoDbInstaller : INeedToInstallSomething
     {
-        Installer installer;
+        readonly Installer installer;
+        readonly IReadOnlySettings settings;
 
-        public DynamoDbInstaller(Installer installer)
+        public DynamoDbInstaller(Installer installer, IReadOnlySettings settings)
         {
             this.installer = installer;
+            this.settings = settings;
         }
 
-        public Task Install(string identity, CancellationToken cancellationToken = default) => installer.Install(cancellationToken);
+        public async Task Install(string identity, CancellationToken cancellationToken = default)
+        {
+            if (!settings.ShouldCreateTables())
+            {
+                return;
+            }
+
+            await installer.CreateOutboxTableIfNotExists(settings.Get<OutboxPersistenceConfiguration>(),
```

Comment:
Maybe only call the installer when the corresponding feature is active? One way to do it is to do it in the saga or outbox feature since they only get activited when the corresponding Core feature is enabled